usecases/vessel-detection/index.md

```diff
@@ -0,0 +1,211 @@
+```
+
+Apply a window to our datacube, and per pixel, update the pixel's value to be the mean of the window surrounding it.
+
+This would be the equivalent of the `data.rolling()` from the xarray example above.
+
+
+```python
+aggregated_window = masked_data.aggregate_spatial_window(size=[30, 30], reducer="mean")
+```
+
+Apply the multiplication of our given value to the result of the spatial window. We multiply to increase the contrast between sea and ship, which makes the detection more reliable.
+
+
+```python
+multiplied = aggregated_window.apply(lambda x: x.multiply(4) )
+```
+
+We then compare the pixel values from the previous node, against the values of the data we initially masked. We want to keep any values from the initially masked data, in order to generate a boolean mask, which should reflect our vessels.
+
+I could not find a nice way to express this in the client, without it re-including the first three nodes of the process graph. So I have applied it directly as a process here.
+
+
+```python
+lt_comparison = multiplied.process("apply", {
+                    "data": multiplied,
+                    "context": masked_data,
+                    "process": {
+                      "process_graph": {
+                        "lt1": {
+                          "process_id": "lt",
+                          "arguments": {
+                            "x": {
+                              "from_parameter": "x"
+                            },
+                            "y": {
+                              "from_parameter": "context"
+                            }
+                          },
+                          "result": True
+                        }
+                      }
+                    }
+                  }
+                )
+```
+
+Convert the resulting boolean datacube to a vector cube, and output the result in save result as a GeoJson. This will make the comparison of our results with the Maritime traffic data more straight forward.
+
+
+```python
+output_data = lt_comparison.process("raster_to_vector", {"data": lt_comparison})
+```
+
+
+```python
+vessel_detection = output_data.save_result(format="GeoJSON")
+```
+
+Create and trigger the job. We can regularly poll the status to ensure it has finished, before moving on to the visualisation of the results.
+
+
+```python
+job_vessel_detection = vessel_detection.create_job(title = "UC2-Vessel-Detection")
+job_vessel_detection.start_job()
+```
+
+
+```python
+job_vessel_detection.status()
+```
+
+
+```python
+job_results = job_vessel_detection.get_results()
+```
+
+
+```python
+# Grab the canonical URL for a given job
+
+def canonical_url_from_job(job: openeo.BatchJob):
+    """ Helper function to get the canonical URL for a finish batch job. """
+    links = [ link for link in job.get_metadata()['links'] if link['rel'] == "canonical" ]
+    return links[0]['href'] if len(links) != 0 else None
+
+canon_url = canonical_url_from_job(job_results)
+canon_url
```

Comment:
It's not so clear what is supposed to happen with canon_url? Does the visualization section depend on it somehow?

Reply:
ok!